src/Orchard.Web/Modules/Orchard.Widgets/Drivers/WidgetPartDriver.cs

```diff
@@ -32,6 +34,14 @@ public class WidgetPartDriver : ContentPartDriver<WidgetPart> {
             widgetPart.AvailableZones = _widgetsService.GetZones();
             widgetPart.AvailableLayers = _widgetsService.GetLayers();
 
+            if (widgetPart.LayerPart == null) {
+                var localizationPart = widgetPart.As<ILocalizableAspect>();
+                ContentItem masterContentItem = localizationPart.MasterContentItem as ContentItem;
+                WidgetPart masterWidgetPart = masterContentItem.Parts.Where(x => x.PartDefinition.Name == "WidgetPart").FirstOrDefault() as WidgetPart;
+                widgetPart.LayerPart = masterWidgetPart.LayerPart;

```

Comment:
Don't assign to the widget. Please create two top level variables, pass them to the shape as properties, and change the view to use these new properties instead of widgetPart.Zone and widgetPart.Layer